src/modules/joke.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file
+ * Modules:
+ *  - {@link joke}
+ * Description:
+ *  Uses the joke api provided by arc to parse a response and send it as a discord embed
+ */
+
+import * as util from '../core/util.js';
+import {botConfig} from '../core/config.js';
+
+// Config manager for nsfw detection
+const jokeConfig = botConfig.modules.joke;
+let JOKE_API_URL = '';
+if (jokeConfig.nsfw) {
+  JOKE_API_URL =
+    'https://v2.jokeapi.dev/joke/Any?blacklistFlags=religious,political,racist,sexist,explicit&type=single';
+} else {
+  JOKE_API_URL =
+    'https://v2.jokeapi.dev/joke/Any?blacklistFlags=nsfw,religious,political,racist,sexist,explicit&type=single';
+}
+
+const joke = new util.RootModule(
+  'joke', // command name
+  'Get a funny joke from the bot', // command description
+  [], // dependencies
+  [], // options
+
+  async (args, interaction) => {
+    fetchJoke()
+      .then(joke => {
+        if (joke == 'Failed to find joke') {
+          util.replyToInteraction(interaction, {
+            embeds: [util.embed.errorEmbed(`${joke}`)],
+          });
+        } else {
+          util.replyToInteraction(interaction, {
+            embeds: [util.embed.infoEmbed(`${joke}`)],
+          });
+        }
+      })
+      .catch(() => {
+        util.replyToInteraction(interaction, {
+          embeds: [util.embed.errorEmbed(`${joke}`)],
+        });
+      });
+  }
+);
+
+async function fetchJoke(): Promise<string> {
+  try {
+    const response = await fetch(JOKE_API_URL);
+    if (response.ok) {
+      const data = await response.json();
+      return data.joke;
+    } else {
+      return 'Failed to find joke';
```

Comment:
Nit:
Rather than returning a string containing the error, it would be preferred to throw an error and let the error be handled elsewhere (where the function is called), and add a jsdoc annotation with `@throws`.
```typescript
/**
 * fetch a joke and return it yada yada
 * @throws Will throw an error if the API call fails
 */
 
// -- snip --

throw new Error('Failed to fetch joke');
 ```

This change will not be required to complete the merge, it is optional.

Reply:
Yeah i'll see if i can do this for ya, it was Ajax that suggested i didn't throw an error in the first place
https://discord.com/channels/1092132405428637886/1092132414228271116/1172493888179667025